### ISSUE TITLE
`stubtest` reports false positives for PyO3 classes after 1.18.1

### ISSUE THREAD

**Issue by extiop** [python/mypy#19838]
Duplicate of https://github.com/python/mypy/issues/19732.

Thanks for the work.

**Comment by brianschubert** [comment]
At a cursory glance, the stubtest errors look correct.  The stubs should define `__new__` instead of `__init__` to match the runtime implementation. (Note how your example mentions `ReaderConfig.__new__()`, not `ReaderConfig.__init__()`)

See https://github.com/python/mypy/issues/19732#issuecomment-3222223351 for more explanation.

**Comment by brianschubert** [comment]
I can confirm that `mla.ReaderConfig` indeed doesn't define `__init__` at runtime, so the stubtest errors are correct:
```python
$ uv run -w mla-archive --python 3.10 python 
>>> import mla
>>> mla.ReaderConfig.__init__ is object.__init__
True
```
#19733 will provide more helpful error messages in situtations like this. Closing as a duplicate of #19732
